build_rust.sh

```diff
@@ -1,6 +1,13 @@
-RUSTFLAGS="--cfg=web_sys_unstable_apis --Z wasm_c_abi=spec" cargo rustc --target=wasm32-unknown-unknown --release -- --emit=llvm-ir
+export RUSTFLAGS="--cfg=web_sys_unstable_apis --Z wasm_c_abi=spec" 
+rustc +nightly \
```

Comment:
This is probably the biggest change so far.

`cargo rustc` injects some defaults flags for the `rustc` invocation, but in the end you were grabbing the `.ll` file anyway, so you while the full crate was still being built (I think), you did not use the product of that work and most of those extra flags do not have an effect on the per-file artifacts.

Instead, I’m invoking `rustc` directly here (with `+nightly` to make sure the right toolchain is used), which allows the same kind of invocation pattern as in `build_c.sh`.

Reply:
`.o` files are actually normal wasm files (you can use `wasm-objdump`) with some extra custom sections for the relocation info.

I was hoping I could build the entire crate/library this way, but sadly I could convice `cargo +nightly build` to spit out a wasm file that `wasm-ld` would accept. 

The upside of that would be that you are much closer to normal rust development, where your options in `Cargo.toml` are being respected etc.

Reply:
Okay lol I just played around and found a way to do this. Ignore this thread (although I’ll leave it open for explanation’s sake), and push another commit.